pkg/attestation/crafter/runners/oidc/gitlab.go

```diff
@@ -0,0 +1,108 @@
+//
+// Copyright 2025 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oidc
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"github.com/rs/zerolog"
+)
+
+// GitlabTokenEnv is the environment variable name for Gitlab OIDC token.
+// #nosec G101 - This is just the name of an environment variable, not a credential
+const GitlabTokenEnv = "GITLAB_OIDC"
+
+// CIServerURLEnv is the environment variable name for Gitlab CI server URL.
+const CIServerURLEnv = "CI_SERVER_URL"
+
+type GitlabToken struct {
+	oidc.IDToken
+
+	// ConfigRefURI is a reference to the current job workflow.
+	ConfigRefURI string `json:"ci_config_ref_uri"`
+
+	// RunnerEnvironment is the environment the runner is running in.
+	RunnerEnvironment string `json:"runner_environment"`
+}
+
+type GitlabOIDCClient struct {
+	Token *GitlabToken
+}
+
+func NewGitlabClient(ctx context.Context, logger *zerolog.Logger) (*GitlabOIDCClient, error) {
+	var c GitlabOIDCClient
+
+	// retrieve the Gitlab server on which the pipeline is running, which is the provider URL
+	providerURL := os.Getenv(CIServerURLEnv)
+	logger.Debug().Str("providerURL", providerURL).Msg("retrieved provider URL")
+	if providerURL == "" {
+		return nil, fmt.Errorf("%s environment variable not set", CIServerURLEnv)
+	}
+
+	tokenContent := os.Getenv(GitlabTokenEnv)
+	logger.Debug().Str("tokenContent", tokenContent).Msg("retrieved token content")
```

Comment:
Printing the token can lead to security issues, please remove "tokenContent" from logs.

Reply:
Mhm, very good catch - removed.